node_settled_handler.go

```diff
@@ -143,24 +173,43 @@ func (p *NodeSettledHandler) handleFinalHtlc(ctx context.Context,
 
 	settledHash, err := p.persister.MarkHtlcSettled(ctx, htlcKey)
 	switch {
-	case err == persistence.ErrHtlcAlreadySettled:
-		return nil
+	case err == persistence.ErrHtlcAlreadyInStatus:
```

Comment:
I see you want to reuse the err message, but this name isn't optimally clear I think. Perhaps `ErrHtlcAlreadyFinal`?